MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2344,7 +2344,7 @@ public int GetNumberOfThreads(EncodingJobInfo state, EncodingOptions encodingOpt
             // Automatic
             if (threads <= 0 || threads >= Environment.ProcessorCount)
             {
-                return 0;
+                return Environment.ProcessorCount;
```

Comment:
`-threads 0` means auto, let ffmpeg decide.
I think threads <= 0 is an invalid value.

Reply:
@nyanmisaka I think threads are ``<= 0`` when, in config, threads are set to ``Auto``. I'll check that.

Maybe this is the most optimal solution?

```
if (threads <= 0)
{
    return 0;
}
else if (threads >= Environment.ProcessorCount) 
{
    return Environment.ProcessorCount;
}
```

Reply:
Btw, not for this PR, but I think we should make server send only the amount of cores that are available, instead of hardcoding them in client.

Reply:
That is most optimal for now.
The upper limit is 8 cores in jf-web, however ffmpeg can utilize 16 cores at most.

Reply:
@nyanmisaka After testing I can confirm you that 0 is returned when web is set to "Auto".

I'll make a commit with the proposed logic alongside the build fixes